src/test/java/org/elasticsearch/index/query/TermQueryBuilderTest.java

```diff
@@ -0,0 +1,106 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.Filter;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.QueryWrapperFilter;
+import org.apache.lucene.search.TermQuery;
+import org.elasticsearch.ElasticsearchNullPointerException;
+import org.elasticsearch.common.lucene.BytesRefs;
+import org.elasticsearch.index.search.child.CustomQueryWrappingFilter;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+
+public class TermQueryBuilderTest extends BaseQueryTestCase<TermQueryBuilder> {
+
+    @Override
+    protected TermQueryBuilder createEmptyQueryBuilder() {
+        return new TermQueryBuilder();
+    }
+
+    /**
+     * @return a TermQuery with random field name and value, optional random boost and queryname
+     */
+
+    protected TermQueryBuilder createTestQueryBuilder() {
+        Object value;
+        switch (randomIntBetween(0, 3)) {
+            case 0: value = randomBoolean(); break;
+            case 1: value = randomAsciiOfLength(8); break;
+            case 2: value = randomInt(10000); break;
+            case 3: value = randomDouble(); break;
+            default: value = randomAsciiOfLength(8);
+        }
+        TermQueryBuilder query = new TermQueryBuilder(randomAsciiOfLength(8), value);
+        if (randomBoolean()) {
+            query.boost(2.0f / randomIntBetween(1, 20));
+        }
+        if (randomBoolean()) {
+            query.queryName(randomAsciiOfLength(8));
+        }
+        return query;
+    }
+
+    @Override
+    protected void assertLuceneQuery(TermQueryBuilder queryBuilder, Query query, QueryParseContext context) throws IOException {
+        assertThat(query, instanceOf(TermQuery.class));
+        assertThat(query.getBoost(), is(queryBuilder.boost()));
+        TermQuery termQuery = (TermQuery) query;
+        assertThat(termQuery.getTerm().field(), is(queryBuilder.fieldName()));
+        assertThat(termQuery.getTerm().bytes(), is(BytesRefs.toBytesRef(queryBuilder.value())));
+        if (queryBuilder.queryName() != null) {
+            Filter namedQuery = context.copyNamedFilters().get(queryBuilder.queryName());
+            assertNotNull(namedQuery);
+            if (namedQuery instanceof CustomQueryWrappingFilter) {
+                assertThat(query, is(((CustomQueryWrappingFilter) namedQuery).getQuery()));
+            } else if (namedQuery instanceof QueryWrapperFilter) {
+                assertThat(query, is(((QueryWrapperFilter) namedQuery).getQuery()));
+            } else {
+                fail("Expected either a QueryWrapperFilter or a CustomQueryWrappingFilter to be registered under " + queryBuilder.queryName() +
+                        " in the query parse context");
+            }
```

Comment:
I wish there was a less impl dependent way of checking that named queries/filters get parsed (without instanceof etc.). Looks good though I don't see many other ways to do this. Btw this is code that every single query should support I believe (apart from match_all?), maybe we should try and make it more generic if possible (not only test wise).